Changes to add Serilog

jernejk opened the pull request:
As per the blog post [TODO: Add URL once the blog post is published], here are the changes.

This PR will intentionally be a draft at all times.

jernejk committed:
Initial implementation of Serilog integration.

jernejk committed:
Minor tweak on csproj.

jernejk committed:
Added Serilog Async integration.

jernejk committed:
Merge pull request #1 from jernejk/features/add-serilog

[WIP] Add Serilog integration

jernejk committed:
Fixed a bug where configuration isn't passed correctly and made sure logs are transfered correctly.

adamchester commented on AspNetCoreSerilogExample.Web/Program.cs:
```diff
@@ -1,17 +1,75 @@
+                Log.Logger.Fatal(e, "Unable to bootstrap web app.");
+
+                // Make sure all the log sinks have processed the last log before closing the application.
+                Log.CloseAndFlush();
```
Wouldn’t it be safer to do this always, rather than only when an exception happens?

jernejk replied:
I agree.
I'm not sure what would happen if the server gracefully shuts down and I'm not sure if Serilog would be correctly disposed.

adamchester commented on AspNetCoreSerilogExample.Web/Program.cs:
```diff
@@ -1,17 +1,75 @@
+                Log.CloseAndFlush();
+            }
+
+            return 0;
```
Why always return 0? Wouldn’t it be better to leave main as a`void`Return type  instead of making the main method return an exit code?

jernejk replied:
Just something that stuck with me from C times. :)

jernejk committed:
Minor improvements based on Adam Chester's suggestions.

jernejk committed:
Added Travis CI yml.

jernejk committed:
Updated Travis yml.

jernejk committed:
Added link to my blog post

jernejk committed:
Updated to .NET Core 3

jernejk committed:
Merge branch 'master' of https://github.com/jernejk/AspNetCoreSerilogExample

jernejk commented:
Due to .NET Core 3.0 use #3 for ASP.NET Core 2.2 delta.

jernejk committed:
Set appsettings.Local.json as required by mistake.

jernejk committed:
Added app.UseSerilogRequestLogging(); which convert plain text HTTP logs to rich logs.

jernejk committed:
Update README.md

Updated readme to include instruction on how to add Serilog to ASP.NET Core 3.0.

jernejk committed:
Update README.md

Changed json type syntax to js

jernejk committed:
Update README.md

jernejk committed:
Update README.md

Appsettings.local.json should be optional.

dependabot-preview[bot] committed:
Bump Serilog.Exceptions from 5.3.1 to 5.4.0

Bumps [Serilog.Exceptions](https://github.com/RehanSaeed/Serilog.Exceptions) from 5.3.1 to 5.4.0.
- [Release notes](https://github.com/RehanSaeed/Serilog.Exceptions/releases)
- [Commits](https://github.com/RehanSaeed/Serilog.Exceptions/compare/5.3.1...5.4.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

jernejk committed:
Update .travis.yml

jernejk committed:
Merge pull request #5 from jernejk/upgrade-travis-build

Update .travis.yml

jernejk committed:
Merge pull request #4 from jernejk/dependabot/nuget/Serilog.Exceptions-5.4.0

Bump Serilog.Exceptions from 5.3.1 to 5.4.0

jernejk committed:
Upgraded to .NET Core 3.1.

jernejk committed:
Updated to .NET 5.

jernejk committed:
Update README.md

jernejk committed:
Updated libraries.

jernejk committed:
Update .travis.yml

jernejk committed:
Added TestController for testing logs.

jernejk committed:
Added MS Logger to compare against Serilog.

jernejk committed:
✨ Upgraded MS Logger project.

jernejk committed:
✨ Upgraded Serilog project to .NET 6.

jernejk committed:
📝 Updated readme.

jernejk committed:
⬆️ Updated to .NET 7.

jernejk committed:
Create .github/workflows/dotnet.yml

jernejk committed:
Update README.md
